libpod/networking_linux.go

```diff
@@ -162,19 +182,92 @@ func (r *Runtime) setupRootlessNetNS(ctr *Container) (err error) {
 			if os.IsTimeout(err) {
 				// Check if the process is still running.
 				var status syscall.WaitStatus
-				_, err := syscall.Wait4(cmd.Process.Pid, &status, syscall.WNOHANG, nil)
+				pid, err := syscall.Wait4(cmd.Process.Pid, &status, syscall.WNOHANG, nil)
 				if err != nil {
 					return errors.Wrapf(err, "failed to read slirp4netns process status")
 				}
+				if pid != cmd.Process.Pid {
+					continue
+				}
 				if status.Exited() || status.Signaled() {
 					return errors.New("slirp4netns failed")
 				}
-
 				continue
 			}
 			return errors.Wrapf(err, "failed to read from slirp4netns sync pipe")
 		}
 	}
+
+	if havePortMapping {
+		const pidWaitTimeout = 60 * time.Second
+		chWait := make(chan error)
+		go func() {
+			interval := 25 * time.Millisecond
+			for i := time.Duration(0); i < pidWaitTimeout; i += interval {
+				// Check if the process is still running.
+				var status syscall.WaitStatus
+				pid, err := syscall.Wait4(cmd.Process.Pid, &status, syscall.WNOHANG, nil)
+				if err != nil {
+					break
+				}
+				if pid != cmd.Process.Pid {
+					continue
+				}
+				if status.Exited() || status.Signaled() {
+					chWait <- errors.New("slirp4netns exited")
```

Comment:
Rather Vague error, can we get any beter?  Can you get the exit code?

Reply:
done